driver-core/src/main/java/com/datastax/driver/core/querybuilder/BuiltStatement.java

```diff
@@ -293,25 +311,6 @@ public Boolean isIdempotent() {
         return !hasNonIdempotentOps();
     }
 
-    @Override
-    public String toString() {
-        try {
-            if (forceNoValues)
-                return getQueryString();
-            // 1) try first with all values inlined (will not work if some values require custom codecs,
-            // or if the required codecs are registered in a different CodecRegistry instance than the default one)
-            return maybeAddSemicolon(buildQueryString(null, CodecRegistry.DEFAULT_INSTANCE)).toString();
```

Comment:
It took me a while to figure that out too. The reason is explained [here](https://github.com/datastax/java-driver/blob/3.1.x/driver-core/src/main/java/com/datastax/driver/core/querybuilder/Utils.java#L222-L224). Basically it is done so to not force users to give an exact type for the parameters, i.e. "bigint" would require longs, "varint" `BigInteger` etc. But once inlined, you could give any `Number` instance and it would work just fine.